search/search-client/src/main/java/io/camunda/search/clients/query/SearchTermQuery.java

```diff
@@ -62,8 +57,13 @@ public Builder caseInsensitive(final Boolean value) {
 
     @Override
     public SearchTermQuery build() {
-      return new SearchTermQuery(
-          Objects.requireNonNull(field), Objects.requireNonNull(value), caseInsensitive);
+      Objects.requireNonNull(field);
+
+      if (value == null || value == TypedValue.NULL || value.isNull() || value.value() == null) {
+        throw new IllegalArgumentException(
+            String.format("Expected non-null value for term query, for field '%s'.", field));
+      }
```

Comment:
For my understanding, was there anything failing with a NPE? Because the transformer transforms the null value into `FieldValue.NULL` that the ES client can handle in a term query.

Reply:
Thanks @romansmirnov I think it was multiple things, at the start. For example, the value was never null, as we always create a TypedValue. I later fixed also that null strings are converted to the Null TypedValue, which we didn't before and then it failed on some unexpected places. 

I think we can remove this and I add a new parameterized test case:

Reply:
Yes, that makes sense to me! Basically, keep the null-check for null Strings and add a new parameterized test case.

Reply:
I just found out that require non null https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T- has an overloaded method that supports messages, I would like to use this one over the other. So I can make sure we add some more context.

Reply:
@romansmirnov I updated the PR